src/request.ts

```diff
@@ -68,14 +68,26 @@ async function startFetchRequest(
     ],
   };
 
-  const shouldRetryRequest = (
+  const shouldRetryRequest = async (
     maxRetry: number,
     resOrErr: Response | Error | FetchError,
-  ): boolean => {
+  ): Promise<boolean> => {
     if (!retryOpts.methods.includes(request.method)) return false;
 
     if (resOrErr instanceof Response) {
-      if (retryOpts.statusCodes.includes(resOrErr.status)) {
+      // REST API status codes: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/errorcodes.htm
+      //
+      // Deleted/expired scratch orgs return 420 and causes a long delay on all requests due to the retry with exponential backoff.
+      // We still want to retry on 420 (Medatata API requests sometimes would return it) so here we'll limit to a maximum of 2 retries.
+      if (resOrErr.status === 420 && resOrErr.headers.get('content-type') === 'text/html') {
+        const salesforceEdgeRes = (await resOrErr.text()).includes('If it takes too long, please contact support or visit our');
+
+        if (salesforceEdgeRes && retryOpts.maxRetries > 0 && retryCount == 2) {
+          return false
+        } else {
+          return true
```

Comment:
This will get into a retry loop here if the `salesforceEdgeRes` text from the server ever changes.
- The status will be `420`
- The content type will be `text/html`
  - Makes it into `if` on `82`
- `salesforceEdgeRes` would be `false` if the server response changes
  - then `shouldRetryRequest` would always return `true`